drivers/public_reports/app/models/public_reports/state_level_homelessness.rb

```diff
@@ -0,0 +1,764 @@
+      homeless_scope.with_service_between(
+        start_date: start_date,
+        end_date: end_date,
+        service_scope: shs_scope,
+      ).
+        joins(:service_history_services).
+        merge(shs_scope).
+        order(shs_t[:date].asc).
+        pluck(cl(she_t[:household_id], she_t[:enrollment_group_id]), shs_t[:age], shs_t[:client_id]).
+        each do |hh_id, age, client_id|
+          next if age.blank? || age.negative?
+
+          key = [hh_id, client_id]
+          households[hh_id] ||= []
+          households[hh_id] << age unless counted_ids.include?(key)
+          counted_ids << key
+        end
+      households
+    end
+    memoize :households
+
+    private def adult_and_child_household_ids(start_date, end_date)
+      adult_and_child_households = Set.new
+      households(start_date, end_date).each do |hh_id, household|
+        child_present = household.any? { |age| age < 18 }
+        adult_present = household.any? { |age| age >= 18 }
+        adult_and_child_households << hh_id if child_present && adult_present
+      end
+      adult_and_child_households
+    end
+    memoize :adult_and_child_household_ids
+
+    private def child_only_household_ids(start_date, end_date)
+      child_only_households = Set.new
+      households(start_date, end_date).each do |hh_id, household|
+        child_present = household.any? { |age| age < 18 }
+        adult_present = household.any? { |age| age >= 18 }
+        child_only_households << hh_id if child_present && ! adult_present
+      end
+      child_only_households
+    end
+    memoize :child_only_household_ids
+
+    private def adult_only_household_ids(start_date, end_date)
+      adult_only_household_ids = Set.new
+      households(start_date, end_date).each do |hh_id, household|
+        child_present = household.any? { |age| age < 18 }
+        # Include clients of unknown age
+        adult_only_household_ids << hh_id unless child_present
+      end
+      adult_only_household_ids
+    end
+    memoize :adult_only_household_ids
+
+    private def total_for(scope, population)
+      count = scope.select(:client_id).distinct.count
+
+      word = case population
+      when :veterans
+        'Veteran'
+      when :adults_with_children, :hoh_from_adults_with_children
+        'Household'
+      else
+        'Person'
+      end
+
+      pluralize(number_with_delimiter(count), word)
+    end
+
+    private def get_us_census_population(race_code: 'All', year:)
+      race_var = \
+        case race_code
+        when 'AmIndAKNative' then NATIVE_AMERICAN
+        when 'Asian' then ASIAN
+        when 'BlackAfAmerican' then BLACK
+        when 'NativeHIOtherPacific' then PACIFIC_ISLANDER
+        when 'White' then WHITE
+        when 'RaceNone' then OTHER_RACE
+        when 'MultiRacial' then TWO_OR_MORE_RACES
+        when 'All' then ALL_PEOPLE
+        else
+          raise "Invalid race code: #{race_code}"
+        end
```

Comment:
Push this up to `GrdaWarehouse::UsCensusApi::Aggregates`? Maybe use a Hash?

Reply:
Maybe, but at the moment that's a module, which would break the API.  I'm going to leave it for right now and trust @yakloinsteak knew what he was doing with this one.

Reply:
I didn't push it to aggregates so as to not mix "census" codes with these codes. Wanted the exception, too. Not sure if you can do that as clearly with a hash (maybe?). You could put the logic in the population method, but it will get complicated if we need more fine-grained values (like age ranges, sex, etc.)